lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -2183,6 +2183,14 @@ Value *PatchInOutImportExport::patchTcsBuiltInInputImport(Type *inputTy, unsigne
       input = builder.getInt32(0);
     break;
   }
+  case BuiltInLayer:
+  case BuiltInViewportIndex: {
+    assert(builtInInLocMap.find(builtInId) != builtInInLocMap.end());
+    const unsigned loc = builtInInLocMap.find(builtInId)->second;
+    auto ldsOffset = calcLdsOffsetForTcsInput(inputTy, loc, nullptr, elemIdx, vertexIdx, builder);
+    input = readValueFromLds(false, inputTy, ldsOffset, builder);
+    break;
+  }
```

Comment:
I believe their handling should be identical to BuiltInPointSize. Please just add the additional `case` statements there.